### ISSUE TITLE
Refresh Token Flow failed in anonymous browsing

### ISSUE THREAD

**Issue by jacksoncastro** [spring-projects/spring-authorization-server#394]
**Describe the bug**

The authorization server generates a session cookie which is blocked during `refresh token`  flow

**To Reproduce**

I have an environment composed of a SPA, a resource server and an authorization server using Spring's new library.
SPA authorizes using the Authorization Code + PKCE flow.

Lib versions:

`spring-boot-starter-parent`: `2.5.2`
`spring-security-oauth2-authorization-server`: `0.1.2`
`Google Chrome`: `92.0.4515.131`

During the `refresh token` flow I have the following problem:

When accessing the `/login` url the authorization server will create the application's session cookie: `JSESSIONID`.
This in turn is created with the type `HttpOnly=true`.

<img width="1107" alt="request" src="https://user-images.githubusercontent.com/3226693/129141430-705e63a1-123c-4c47-90e3-12fd5fd4ec4c.png">

1. The client starts the flow by requesting the authorization server (`/oauth2/token`)
1. The browser does not send the session cookie claiming the following:
`This cookie didn't specify a "SameSite" attribute when it was stored and was defaulted to "SameSite=Lax," and was blocked because the​ request was made from a different site and was not initiated by a top-level navigation. The cookie had to have been set with "SameSite=None" to enable cross-site usage.`
1. The `org.springframework.security.oauth2.server.authorization.web.authentication.OAuth2RefreshTokenAuthenticationConverter` converter invalidates the request as there is no session (the browser did not send). And the server redirects to `/login`.

Basically, the browser asks that this cookie has the following attributes to allow its use:

`SameSite=None;Secure=true;`

*Note*: This blocking is by default in browser settings.

**Expected behavior**

Get access token from `refresh token` flow.

**Comment by jgrandja** [comment]
@jacksoncastro 

It's not clear to me the issue you are having. It seems that the SPA was able to obtain the initial access token:

> SPA authorizes using the Authorization Code + PKCE flow

But it fails when it tries to refresh the access token after it expires.

If this is a CORS issue then you wouldn't be able to obtain the access token to begin with since the `/oauth2/token` endpoint is called in both flows.

To better understand your application setup and flow, please provide a [minimal sample](https://stackoverflow.com/help/minimal-reproducible-example) with detailed steps on how to reproduce.

**Comment by spring-projects-issues** [comment]
Closing due to lack of requested feedback. If you would like us to look at this issue, please provide the requested information and we will re-open the issue.
